src/VehicleSelect/VehicleSelect.tsx

```diff
@@ -0,0 +1,221 @@
+              }
+            ]);
+          }}
+          value={selectedProject}
+        />
+      </Box>
+      <Box flex="40%">
+        <Autocomplete
+          disabled={selectedProject === ""}
+          label="Model Year"
+          required
+          multiple={false}
+          options={allModelYears}
+          onChange={(_event, value) => {
+            const newValue = value === null ? "" : value;
+            onVehicleChange([
+              {
+                _id: "",
+                gate: "",
+                modelYear: newValue,
+                project: selectedProject,
+                variant: ""
+              }
+            ]);
+          }}
+          value={selectedModelYear}
+        />
+      </Box>
+      <Box flex="40%">
+        <Autocomplete
+          disabled={selectedModelYear === ""}
+          label="Vehicle Variant"
+          required
+          multiple={true}
+          options={allVariants}
+          onChange={(_event, value) => {
+            const newVehicles = allVehicles.filter(
+              v =>
+                v.projectCode === selectedProject &&
+                v.modelYear === selectedModelYear &&
+                value.includes(v.variant)
+            );
+            if (newVehicles.length === 0) {
+              onVehicleChange([
+                {
+                  _id: "",
+                  gate: "",
+                  modelYear: selectedModelYear,
+                  project: selectedProject,
+                  variant: ""
+                }
+              ]);
+              return;
+            }
+            if (selectedGates.length === 0) {
+              onVehicleChange(
+                newVehicles.map(v => ({
+                  _id: v._id,
+                  gate: "",
+                  modelYear: v.modelYear,
+                  project: v.projectCode,
+                  variant: v.variant
+                }))
+              );
+            }
+            if (selectedGates.length > 0) {
+              const newVehiclesWithGates = selectedGates.flatMap(gate =>
+                newVehicles.map(v => ({
+                  _id: v._id,
+                  gate,
+                  modelYear: v.modelYear,
+                  project: v.projectCode,
+                  variant: v.variant
+                }))
+              );
+              onVehicleChange(newVehiclesWithGates);
+            }
+          }}
+          value={selectedVariants}
+        />
+      </Box>
+      <Box flex="40%">
+        <Autocomplete
+          disabled={selectedVariants.length === 0}
+          required
+          multiple={true}
+          label="Gate"
+          options={allGates}
+          onChange={(_event, value) => {
+            const newVehicles = allVehicles.filter(
+              v =>
+                v.projectCode === selectedProject &&
+                v.modelYear === selectedModelYear &&
+                selectedVariants.includes(v.variant)
+            );
+            if (value.length === 0) {
```

Comment:
please add comments for the conditions you are checking

Reply:
addded